main.go

```diff
@@ -8,14 +8,21 @@ import (
 	"net/http"
 	"os"
 
+	"github.com/sourcegraph/go-vcsurl"
 	"github.com/tcard/navpatch/internal"
 	"github.com/tcard/navpatch/navpatch"
+	"github.com/tcard/navpatch/navpatch/repositories"
 )
 
 func main() {
 	listenAddr, baseDir, rawPatch := processArgs()
 
-	nav, err := navpatch.NewNavigator(baseDir, rawPatch)
+	r, err := buildRepository(baseDir)
+	if err != nil {
+		internal.ErrorExit(err)
+	}
+
+	nav, err := navpatch.NewNavigator(r, rawPatch)
```

Comment:
`buildRepository` should be done inside `NewNavigator`. The command should be a very thin wrapper for the library. Users of navpatch as a library also want that convenience. This kind of "stringly-typed" interface is commonplace in the standard library.